tags.go

```diff
@@ -99,25 +102,35 @@ func ValidateTagKey(key string) error {
 	return nil
 }
 
-func (client *Client) AssignTags(identifier string, tags map[string]string) ([]Tag, error) {
+func (client *Client) AssignTagsWithTags(identifier string, tags []Tag) ([]Tag, error) {
```

Comment:
Don't love the function name.  A customer with little context would read this and not really understand it.  This is why i recommended `AssignTagsPro` or `AssignTagsExt`.  You could even then add a function definition string like `// AssignTagsExt - Is AssignTags using a []Tag instead of map[string]string`